### DISCUSSION TITLE
History menu quirks

### DISCUSSION THREAD

**Discussion by chrisr87** [marlonrichert/zsh-autocomplete#585] [General]
I just updated to the latest and this behavior is still the same so I'm wondering if it's intended to be this way. It seems to be a recent change within the last few months.

But what is the correct way to use the history menu? I press <kbd>↑</kbd> to trigger the history menu. And I use <kbd>Ctrl + G</kbd> to cancel out of it without using any of the history entries. And while it closes the list of history entries, it seems to still be stuck in history mode because when I press <kbd>↓</kbd>, I expect to see directories but I see the history menu again. Just wondering if this is a regression or if it's intentional.

Also, <kbd>←</kbd> and <kbd>→</kbd> moves up and down through the history entries. Maybe, I'm remembering wrong, but it used to select the entry and back to the prompt. How do I change this back? It seems to be redundant to have all the arrows do the same thing in the history menu. It's more useful for me if the <kbd>←</kbd>, for example, uses the selected entry and just moves left on the prompt.

**Comment by marlonrichert** [comment]
Moved to issue #586.
